qiskit/transpiler/preset_passmanagers/level2.py

```diff
@@ -242,18 +242,10 @@ def _unroll_condition(property_set):
         optimization = plugin_manager.get_passmanager_stage(
             "optimization", optimization_method, pass_manager_config, optimization_level=2
         )
-    if scheduling_method is None or scheduling_method in {"alap", "asap"}:
-        sched = common.generate_scheduling(
-            instruction_durations,
-            scheduling_method,
-            timing_constraints,
-            inst_map,
-            target=target,
-        )
-    else:
-        sched = plugin_manager.get_passmanager_stage(
-            "scheduling", scheduling_method, pass_manager_config, optimization_level=2
-        )
+    
+    sched = plugin_manager.get_passmanager_stage(
+        "scheduling", scheduling_method, pass_manager_config, optimization_level=2
+    )
```

Comment:
I think we need to preserve the `scheduling_method is None` path here. I believe the error here is we don't have a plugin when `scheduling_method=None` (i.e. the default).  You can probably do something like:

```suggestion
    if scheduling_method is None:
        sched = common.generate_scheduling(
            instruction_durations,
            scheduling_method,
            timing_constraints,
            inst_map,
            target=target,
        )
    else:
        sched = plugin_manager.get_passmanager_stage(
            "scheduling", scheduling_method, pass_manager_config, optimization_level=2
        )
```

We still need to call `generate_scheduling` because there are passes that might need to run in the scheduling stage even if no method is specified IIRC.